src/index.ts

```diff
@@ -62,53 +63,112 @@ async function submitToDataDog(
+  const response = await got(commitUrl, {
     headers: { Authorization: `token ${githubToken}` },
+    responseType: 'json',
   });
-  return await response.json();
+  return response.body;
+}
+
+function parseTimestamp(timestamp: string) {
+  return Math.floor(new Date(timestamp).getTime() / 1000);
+}
+
+function getBranch(webhookPayload: WebhookPayload) {
+  switch (webhookPayload.action) {
+    case 'push':
+      return webhookPayload.after;
+    case 'opened':
+    case 'edited':
+    case 'closed':
+    case 'assigned':
+    case 'unassigned':
+    case 'review_requested':
+    case 'review_request_removed':
+    case 'ready_for_review':
+    case 'converted_to_draft':
+    case 'labeled':
+    case 'unlabeled':
+    case 'synchronize':
+    case 'auto_merge_enabled':
+    case 'auto_merge_disabled':
+    case 'locked':
+    case 'unlocked':
+    case 'reopened':
+      return webhookPayload.pull_request?.head.ref;
+    default:
+      return undefined;
+  }
+}
+
+function getCommitId(webhookPayload: WebhookPayload) {
+  switch (webhookPayload.action) {
+    case 'push':
+      return webhookPayload.sha;
+    case 'opened':
+    case 'edited':
+    case 'closed':
+    case 'assigned':
+    case 'unassigned':
+    case 'review_requested':
+    case 'review_request_removed':
+    case 'ready_for_review':
+    case 'converted_to_draft':
+    case 'labeled':
+    case 'unlabeled':
+    case 'synchronize':
+    case 'auto_merge_enabled':
+    case 'auto_merge_disabled':
+    case 'locked':
+    case 'unlocked':
+    case 'reopened':
+      return webhookPayload.pull_request?.head.sha;
+    default:
+      return undefined;
+  }
 }
 
 async function reportCountOfFilesConverted(
   sourcePath: string,
-  webhookPayload: WebhookPayload,
+  commit: any,
+  branch: string,
   datadogMetric: string,
   datadogApiKey: string,
-  githubToken: string,
 ) {
-  const response = await getData(webhookPayload.head_commit.id, githubToken);
   try {
-    const { stdout, stderr } = await exec(`npx --quiet cloc --include-lang=TypeScript,JavaScript --json ${sourcePath}`);
-    if (stderr) {
-      throw new Error(stderr);
-    }
-
-    const renamedFiles = response.files
-      ? response.files.filter((f: { previous_filename?: string }) => f.previous_filename)
+    const renamedFiles = commit.files
+      ? commit.files.filter((f: { previous_filename?: string }) => f.previous_filename)
       : [];
-    const otherFiles = response.files ? response.files.filter((f: { filename?: string }) => f.filename) : [];
+    const otherFiles = commit.files ? commit.files.filter((f: { filename?: string }) => f.filename) : [];
     const count = findFileCountOfJSConversionsToTS(renamedFiles);
     const otherCount = findFileCountOfJSConversionsToTSForAllFiles(otherFiles);
     const totalCount = count + otherCount;
 
-    //do not report 0 counts
+    // do not report 0 counts
     if (totalCount === 0) {
+      core.info('Ignoring commit with no file changes');
```

Comment:
We're ignoring commits with no JS or TS/X file changes.

Reply:
updated, thanks